lib/module/src/module.rs

```diff
@@ -123,9 +123,18 @@ pub enum ModuleError {
     /// Indicates an identifier was used before it was declared
     #[fail(display = "Undeclared identifier: {}", _0)]
     Undeclared(String),
-    /// Indicates an identifier was used contrary to the way it was declared
+    /// Indicates an identifier was used as data/function first, but then used as the other
     #[fail(display = "Incompatible declaration of identifier: {}", _0)]
     IncompatibleDeclaration(String),
+    /// Indicates a function identifier was declared with a
+    /// different signature than declared previously
+    #[fail(
+        display = "Incompatible declaration of function identifier: {} ({:?} vs {:?})",
```

Comment:
Can we word the error message to make it clear which `{:?}` is the existing declaration and which is the new one?

Perhaps something like "Function {} signature {:?} is incompatible with previous declaration {:?}"?